Source/Core/DolphinWX/GameListCtrl.cpp

```diff
@@ -237,6 +238,9 @@ static bool ShouldDisplayGameListItem(const GameListItem& item)
   if (!show_platform)
     return false;
 
+  if (item.GetPlatform() == DiscIO::Platform::WII_WAD && !IOS::ES::IsChannel(item.GetTitleID()))
+    return false;
```

Comment:
Hmm... Doing it the way I suggested means that we have to re-scan invalid WADs each time the game list is refreshed, though... Maybe it's better to do it the way it's done in this PR.